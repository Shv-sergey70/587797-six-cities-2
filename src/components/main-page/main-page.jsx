import React from 'react';
import PropTypes from "prop-types";
import {offerPropTypes} from "../card/card";
import {connect} from "react-redux";
import Selectors from '../../selector';
import PageHeader from '../page-header/page-header';
import Operation from "../../operation";
import PlacesPage from "../places-page/places-page";
import PlacesPageEmpty from "../places-page-empty/places-page-empty";

class MainPage extends React.PureComponent {
  componentDidMount() {
    const {
      loadOffers
    } = this.props;

    loadOffers();
  }

  render() {
    const {
      currentOffers
    } = this.props;

    return <div className="page page--gray page--main">
      <PageHeader/>

      {currentOffers.length > 0 ? <PlacesPage/> : <PlacesPageEmpty/>}
    </div>;
  }
}

MainPage.propTypes = {
  currentOffers: PropTypes.arrayOf(offerPropTypes),
  loadOffers: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentOffers: Selectors.getCurrentOffers(state)
});

const mapDispatchToProps = (dispatch) => ({
  loadOffers: () => dispatch(Operation.loadOffers())
});

export {MainPage};

export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
