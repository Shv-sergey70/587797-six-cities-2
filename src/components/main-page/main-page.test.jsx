import React from 'react';
import {MainPage} from "./main-page";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

export const cities = [
  {
    name: `Paris`
  },
  {
    name: `Brussels`
  },
  {
    name: `Amsterdam`
  },
];

const offers = [
  {
    id: 1,
    city: cities[0],
    title: `Beautiful & luxurious apartment at great location`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    type: `Apartment`,
    price: 80,
    rating: 4.5,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    }
  },
  {
    id: 2,
    city: cities[1],
    title: `Wood and stone place`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: false,
    type: `Private room`,
    price: 180,
    rating: 4,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    }
  },
  {
    id: 3,
    city: cities[0],
    title: `Canal View Prinsengracht`,
    previewImage: `/img/apartment-02.jpg`,
    isPremium: false,
    isFavorite: true,
    type: `Apartment`,
    price: 120,
    rating: 5,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    }
  },
  {
    id: 4,
    city: cities[2],
    title: `Nice, cozy, warm big bed apartment`,
    previewImage: `/img/apartment-03.jpg`,
    isPremium: false,
    isFavorite: false,
    type: `Private room`,
    price: 40,
    rating: 3,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    }
  }
];

it(`Main-page correctly renders`, () => {
  const tree = shallow(
      <MainPage
        allOffers = {offers}
        currentCity = {
          {
            name: `Paris`
          }
        }
        currentOffers = {offers.filter((offer) => offer.city.name === `Paris`)}
        onCityClick = {() => {}}
      />);

  expect(toJson(tree)).toMatchSnapshot();
});
