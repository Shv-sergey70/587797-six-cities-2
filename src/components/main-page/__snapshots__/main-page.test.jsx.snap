// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main-page correctly renders 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <WithActiveItem
          cities={
            Array [
              Object {
                "name": "Paris",
              },
              Object {
                "name": "Brussels",
              },
              Object {
                "name": "Amsterdam",
              },
            ]
          }
          currentCity={
            Object {
              "name": "Paris",
            }
          }
          defaultActiveItem="Paris"
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                >
                   
                </use>
              </svg>
            </span>
            <ul
              className="places__options places__options--custom places__options--opened"
            >
              <li
                className="places__option places__option--active"
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <WithActiveItem
            defaultActiveItem={0}
            offers={
              Array [
                Object {
                  "city": Object {
                    "name": "Paris",
                  },
                  "id": 1,
                  "isFavorite": true,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                  },
                  "previewImage": "/img/apartment-01.jpg",
                  "price": 80,
                  "rating": 4.5,
                  "title": "Beautiful & luxurious apartment at great location",
                  "type": "Apartment",
                },
                Object {
                  "city": Object {
                    "name": "Paris",
                  },
                  "id": 3,
                  "isFavorite": true,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                  },
                  "previewImage": "/img/apartment-02.jpg",
                  "price": 120,
                  "rating": 5,
                  "title": "Canal View Prinsengracht",
                  "type": "Apartment",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              offersLocations={
                Array [
                  Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                  },
                  Object {
                    "latitude": 52.3909553943508,
                    "longitude": 4.85309666406198,
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
